src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty() || !HashUtil.hashPassword(password,
+                userFromDbOptional.get().getSalt())
+                .equals(userFromDbOptional.get().getPassword())) {
+            throw new AuthenticationException("Can't login with input data. Email: " + email);
+        }
+        return userFromDbOptional.get();
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (email.length() < 4) {
```

Comment:
Why do you need 2 conditional email.length() < 4 and email.length() < 6 ? 
Maybe you shouild leave only one

Reply:
Just leave this password.length() < 6.